### DISCUSSION TITLE
Class 'MockSomeAsyncNotifier' has no instance method '_setElement'.

### DISCUSSION THREAD

**Discussion by DaVaMa1** [rrousselGit/riverpod#2275] [Q&A] [answered]
I have a few AsyncNotifiers in my project. One of those get's stubbed out by a Mockito mock during a test. While upgrading from Riverpod 2.1.1 to 2.3.0 my tests started failing. 

I've recently inherited this project, and am very new to Riverpod. 

Am I not supposed to mock out the AsyncNotifier? Anyone else had this happen?

**Comment by rrousselGit** [comment, marked as answer]
Do `class YourMock extends AsyncNotifier with Mock implements YourNotifier` (or `extends AnotherNotifierType` if your notifier extends something else)

**Reply by DaVaMa1** [reply]
Thanks, that worked.
